themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,339 @@ variables:
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`);
+
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
+    // delete the policy statements
+    policy.Statement = [];
+    return policy;
+});
+```
+
+For more details [view the NodeJS documentation](/docs/reference/pkg/nodejs/pulumi/pulumi/).
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const jsonIAMPolicy = pulumi.output(`{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`);
+
+const policyWithNoStatements = pulumi.jsonParse(jsonIAMPolicy).apply(policy => {
+    // delete the policy statements
+    policy.Statement = [];
+    return policy;
+});
+```
+
+For more details [view the NodeJS documentation](/docs/reference/pkg/nodejs/pulumi/pulumi/).
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+json_IAM_policy = pulumi.Output.from_input('''
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}
+''')
+
+def update_policy(policy):
+    # delete the policy statements
+    policy.update({'Statement': []})
+    return policy
+
+policy_with_no_statements = \
+    pulumi.Output.json_loads(json_IAM_policy).apply(update_policy)
+```
+
+For more details [view the Python documentation](/docs/reference/pkg/python/pulumi/).
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+jsonIAMPolicy := pulumi.ToOutput(`{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}`).(pulumi.StringInput)
+
+policyWithNoStatements := pulumi.JSONUnmarshal(jsonIAMPolicy).ApplyT(
+    func(v interface{}) (interface{}, error) {
+
+    // delete the policy statements
+    v.(map[string]interface{})["Statement"] = []pulumi.ArrayOutput{}
+    return v, nil
+})
```

Comment:
The function body needs another level of indentation.

```suggestion
        // delete the policy statements
        v.(map[string]interface{})["Statement"] = []pulumi.ArrayOutput{}
        return v, nil
    })
```

Also, shouldn't this be `[]interface{}`, or `[]ArrayOutput`?